pages/contacts/[id]/index.js

```diff
@@ -32,30 +33,22 @@ const StyledInfo = styled.p`
   padding: 1rem;
   border-radius: 8px;
 `;
-
+//----------------------------------------------- FUNCTION------------HERE
 export default function ContactDetail() {
   const router = useRouter();
-  const { dynamicId } = router.query;
-
-  const contact = contacts.find((contact) => contact.id === dynamicId);
-
-  if (!contact) {
-    return <p>Kontakte nicht gefunden</p>;
-  }
+  const { query } = router;
+  const { id } = query;
+  const { data: contact, isLoading, error } = useSWR(`/api/contacts/${id}`);
 
-  const { name, photo, phone, note } = contact;
+  if (isLoading) return <h2>Loading...</h2>;
+  if (error) return <h2>Error...</h2>;
 
   return (
     <>
       <StyledHeader>
         <StyledHeaderContainer>
-          <StyledH3>{name}</StyledH3>
-          <StyledImage
-            src={photo}
-            width={34}
-            height={34}
-            alt={`Photo of ${name}`}
-          />
+          <StyledH3>{contact.name}</StyledH3>
+          <GrayCirclePhotoPlaceholder />
         </StyledHeaderContainer>
       </StyledHeader>
       <StyledHR />
```

Comment:
As an `hr` element is usually used to add a break between `p` elements [(see MDN for further reference)](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/hr), I think it might be better to work with `border-bottom` in your `StyledHeader` component from a semantic point of view.

Reply:
Thank you for your thoughtful comments.

I would like to keep the HR as a styling component, as I use it and will use it very often in the future in many other components. I think that putting a border button in a correct position on each page is "less" clean than a standard separation for the whole application (or placeholder?). Please let me keep this separation, as it not only helps to separate visually but also helps to recognize visually in the component how it is structured.   Please let me leave it at that, i promise you, when the application is more or less finished, I can transfer this code to the border-botton and refine my semantics and styles, but right now I think it's too early.

Reply:
This sound reasonable to me. If you want to think about this later this is alright for me. :)